src/integrationtests/java/util/IntegrationTestBase.java

```diff
@@ -0,0 +1,205 @@
+                        TimeUnit.MILLISECONDS.sleep(firstEventWaitTime);
+                    } catch (InterruptedException e) {
+                        // TODO Auto-generated catch block
+                        LOGGER.error(e.getMessage(), e);
+                    }
+                    alreadyExecuted = true;
+                }
+            }
+
+            // wait for all events to be processed
+            waitForEventsToBeProcessed(eventsCount);
+            checkResult(getCheckData());
+        } catch (IOException e) {
+            LOGGER.error(e.getMessage(), e);
+        }
+    }
+
+    /**
+     * @return path to file with rules, that is used in flow test
+     */
+    protected abstract String getRulesFilePath();
+
+    /**
+     * @return path to file, with events, that is used in flow test
+     */
+    protected abstract String getEventsFilePath();
+
+    /**
+     * @return list of event names, that will be used in flow test
+     */
+    protected abstract List<String> getEventNamesToSend();
+
+    /**
+     * @return map, where key - _id of expected aggregated object value - expected
+     *         aggregated object
+     *
+     */
+    protected abstract Map<String, JsonNode> getCheckData() throws IOException;
+
+    protected JsonNode getJSONFromFile(String filePath) throws IOException {
+        String expectedDocument = FileUtils.readFileToString(new File(filePath), "UTF-8");
+        return objectMapper.readTree(expectedDocument);
+    }
+
+    // count documents that were processed
+    private long countProcessedEvents(String database, String collection) {
+        MongoClient mongoClient = null;
+        mongoClient = mongoDBHandler.getMongoClient();
+        MongoDatabase db = mongoClient.getDatabase(database);
+        MongoCollection table = db.getCollection(collection);
+        return table.count();
+    }
+
+    protected void waitForEventsToBeProcessed(int eventsCount) {
+        // wait for all events to be processed
+        long processedEvents = 0;
+        while (processedEvents < eventsCount) {
+            processedEvents = countProcessedEvents(database, event_map);
+            LOGGER.info("Have gotten: " + processedEvents + " out of: " + eventsCount);
+            try {
+                TimeUnit.MILLISECONDS.sleep(10000);
+            } catch (InterruptedException e) {
+                LOGGER.error(e.getMessage(), e);
+            }
+        }
+    }
+
+    private void checkResult(final Map<String, JsonNode> checkData) {
+        checkData.forEach((id, expectedJSON) -> {
+            try {
+                String document = objectHandler.findObjectById(id);
+                JsonNode actualJSON = objectMapper.readTree(document);
+                LOGGER.info("Complete aggregated object: " + actualJSON);
+                JSONAssert.assertEquals(expectedJSON.toString(), actualJSON.toString(), false);
+            } catch (IOException | JSONException e) {
+                LOGGER.error(e.getMessage(), e);
```

Comment:
you get much cleaner code if you throw the exception here. The logging does not help too much anyway.

Reply:
done